enricher/api/src/main/java/io/fabric8/maven/enricher/api/EnricherContext.java

```diff
@@ -109,8 +111,12 @@ default boolean hasDependency(String groupId, String artifactId) {
         return Optional.empty();
     }
 
-    // ===========================================================================================
-    // To be removed:
 
-    OpenShiftDependencyResources getOpenshiftDependencyResources();
+    PlatformMode getPlatformMode();
+
+    RuntimeMode getRuntimeMode();
+
+    Object getProperty(String key);
+
+    Properties getProperties();
```

Comment:
What's about having only one of these methods to keep the API slim? 

Alternatively, you could make `getProperty()` a "default" interface method, too.

And don't forget to add JavaDoc. While I wouldn't insist on having docs everywhere, for public API which are used e.g. for implementing Enrichers, good documentation is mandatory.